lib/simp/rake/beaker.rb

```diff
@@ -4,6 +4,31 @@
 require 'fileutils'
 require 'puppetlabs_spec_helper/rake_tasks'
 
+unless File.exists?('metadata.json')
+  # This is something just using beaker that wants the special sauce. In this
+  # case, we really don't want things that only have to do with puppet modules.
+  keep_tasks = [
+    '^spec$',
+    '^spec_',
+    '^rubocop',
+    '^parallel_',
+    '^beaker'
+  ].map{|x| Regexp.new(x)}
+
+  Rake::Task.tasks.each do |task|
+    unless keep_tasks.any?{ |regexp| regexp.match?(task.name) }
```

Comment:
The whitelist implementation doesn't do what it says—it doesn't remove "things that only have to do with puppet modules," it silently removes _anything_ that isn't spec, rubocop, or beaker. 

Among other things, it would also remove tasks like **`clean`**, **`default`**, **`gitlab:ci_lint`**, simp-adapter's **`log_pkg_rpm`**, simp-rsync's **`check_facl`**, simp cli's **`chmod`** and **`pkg:gem`**, simp-gpgkey's **`fingerprint`** and **`download`**, and almost anything you'd be likely to put in a `rakelib/` directory.

Reply:
@op-ct Where this is called, it only modifies tasks that have been created up to that point. It shouldn't touch anything that doesn't match one of the regular expressions.